src/pages/cache/purge-cache/PurgeCache.jsx

```diff
@@ -0,0 +1,144 @@
+import { projectModules } from "../../../constants";
+import RadioCards from "../../../components/radio-cards/RadioCards"
+import FormItemLabel from "../../../components/form-item-label/FormItemLabel"
+import ConditionalFormBlock from "../../../components/conditional-form-block/ConditionalFormBlock";
+import { getDbConfigs, loadDbConfig, getTrackedCollections, loadDbRules, loadDbSchemas } from "../../../operations/database";
+import { notify, decrementPendingRequests, incrementPendingRequests } from "../../../utils";
+import { purgeCache } from "../../../operations/cache";
+
+const PurgeCache = () => {
+
+    const { projectID } = useParams();
+
+    useEffect(() => {
+        if (projectID) {
+            incrementPendingRequests()
+            loadDbConfig(projectID)
+                .catch(ex => notify("error", "Error fetching database config", ex))
+                .finally(() => decrementPendingRequests())
+
+            incrementPendingRequests()
+            loadDbSchemas(projectID)
+                .catch(ex => notify("error", "Error fetching database schemas", ex))
+                .finally(() => decrementPendingRequests())
+
+            incrementPendingRequests()
+            loadDbRules(projectID)
+                .catch(ex => notify("error", "Error fetching database rules", ex))
+                .finally(() => decrementPendingRequests())
+        }
+    }, [])
+
+    const [form] = Form.useForm();
+    const [selectedDb, setSelectedDb] = useState("");
+    const [value, setValue] = useState("");
+    const dbConfigs = useSelector(state => getDbConfigs(state))
+    const trackedCollections = useSelector(state => getTrackedCollections(state, selectedDb))
+
+    const dbList = Object.keys(dbConfigs)
+
+    const handleFinish = (values) => {
+        let data = {};
+        switch (values) {
+            case "all":
+                data = values;
+                break;
+            case "database":
+                data = {
+                    level: "database",
+                    options: {
+                        db: values.db,
+                        col: "*"
+                    }
+                }
+                break;
+            case "table":
+                data = {
+                    level: "database",
+                    options: {
+                        db: values.db,
+                        col: values.col
+                    }
+                }
+                break;
+            default:
+                break;
+        }
+        incrementPendingRequests()
+        purgeCache(projectID, data)
+        .then(() => notify("success", "Success", "Cache purged successfully"))
+        .catch(ex => notify("error", "Error", ex))
+        .finally(() => decrementPendingRequests())
+    }
+
+    const handleSelectDatabase = value => setSelectedDb(value)
+    const handleSearch = value => setValue(value);
```

Comment:
Are these functions really required? We can directly use the setter functions as it is as these handler functions contains no extra logic